lib/sles4sap/sap_deployment_automation_framework/networking.pm

```diff
@@ -334,7 +334,16 @@ sub assign_defined_network {
 
         # Attempt to acquire network file lease (update modification time)
         # This will prevent other tests from spending time checking this file since it is already taken.
-        return if !acquire_network_file_lease(network_lease_file => $lease_file);
+        if (!acquire_network_file_lease(network_lease_file => $lease_file)) {
+            $lease_file = 0;
+            if ($count > $num) {
+                record_info("Debug: searched for $count times, more than $num times, return to create new network");
+                return;
+            } else {
+                record_info("Debug: searched for $count times, less than $num times, continue to search old network");
+                next;
```

Comment:
I am wondering if `next` is needed here in the end. Since after if statement ends the code goes straight to the next. loop. 
But no need to change it.

Reply:
Got it.